### ISSUE TITLE
[Pointcut] Add support for matching by return-type

### ISSUE THREAD

**Issue by lisachenko** [goaop/framework#370]
For PHP7 it's allowed to use return typehint for methods and functions. Framework should support this kind of syntax via pointcut expression.

Proposed format is following:
```
execution(public Example->method(*): string)
execution(Demo\*\Test\**\*(*): bool)
```

First one for methods and second one for functions.

**Comment by lisachenko** [comment]
Need to implement matching in PhpStorm plugin too, issue https://github.com/goaop/idea-plugin/issues/17

**Comment by lisachenko** [comment]
Implemented in #371
